### PR TITLE
Addon-docs: Fix React Suspense in source snippets.

### PR DESCRIPTION
**Pull request by zhyd1997**
Issue: #11554

## What I did

add `<React.Suspense>` support in `source snippets.

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by zhyd1997** [comment]
Refactored PR: https://github.com/storybookjs/storybook/pull/19785

File: code/renderers/react/src/docs/jsxDecorator.test.tsx [hunk 1]
```diff
@@ -155,6 +155,23 @@ describe('renderJsx', () => {
...
+        <React.Suspense fallback={null}>
+          <div>{props.children}</div>
+        </React.Suspense>
+      );
+    }
+
+    expect(renderJsx(createElement(SuspenseComponent, {}, 'I am Suspense'), {}))
+      .toMatchInlineSnapshot(`
+      <SuspenseComponent>
+        I am Suspense
```
Comment (shilman) [inline code/renderers/react/src/docs/jsxDecorator.test.tsx:170]: Any ideas why the `<div>` is not showing up?

**Review by shilman — COMMENTED** [review]
Thanks for fixing this @zhyd1997 !! Is there a full list of these "known symbols" somewhere & can we do all of them in this PR?

File: code/renderers/react/src/docs/jsxDecorator.test.tsx (hunk 1 above)
Comment (zhyd1997) [reply code/renderers/react/src/docs/jsxDecorator.test.tsx:170]: Hi @shilman 

I submitted a refactor PR: #19785 to handle other "know symbols", but have some issues on CI, you can check it.

Comment (zhyd1997) [reply code/renderers/react/src/docs/jsxDecorator.test.tsx:170]: Uhh, `div`, sorry, it's my problem, I will add it when I come back my office.

Comment (zhyd1997) [reply code/renderers/react/src/docs/jsxDecorator.test.tsx:170]: Actually It works for new creat-react-app when I added them in bundled code, I will figure out why the test passed with wrong snapshot.

**Comment by zhyd1997** [comment]
Hi @shilman!

with the help of @JReinhold on Discord, I have fixed the CI errors on PR #19785 

- added a new react type support -- `StrictMode`, other types will be added later, and
- in my opinion, the `div` missing is another issue...

so could we talk about the feature/bug on PR #19785 ?

**Comment by ndelangen** [comment]
@zhyd1997 Is the other PR a replacement for this one?

Can I close this PR, then?

**Comment by zhyd1997** [comment]
Hi @ndelangen 

The replacement: https://github.com/storybookjs/storybook/pull/19785

**Pull request closed without merging**